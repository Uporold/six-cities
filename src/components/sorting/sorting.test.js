import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import Sorting from "./sorting";
import NameSpace from "../../redux/name-space";

const mockStore = configureStore([]);

it(`Should Places List render correctly`, () => {
  const store = mockStore({
    [NameSpace.APP]: {
      currentSortType: `Popular`,
    },
  });
  const tree = renderer
    .create(
      <Provider store={store}>
        <Sorting />
      </Provider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
