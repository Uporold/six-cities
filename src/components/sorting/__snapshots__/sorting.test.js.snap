// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Places List render correctly 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by
  </span>
  <span
    className="places__sorting-type"
    onClick={[Function]}
    tabIndex="0"
  >
    Popular
    <svg
      className="places__sorting-arrow"
      height="4"
      style={null}
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom "
    onClick={[Function]}
  >
    <li
      className="places__option places__option--active"
      data-sort-type="Popular"
      tabIndex="0"
    >
      Popular
    </li>
    <li
      className="places__option "
      data-sort-type="Price: low to high"
      tabIndex="0"
    >
      Price: low to high
    </li>
    <li
      className="places__option "
      data-sort-type="Price: high to low"
      tabIndex="0"
    >
      Price: high to low
    </li>
    <li
      className="places__option "
      data-sort-type="Top rated first"
      tabIndex="0"
    >
      Top rated first
    </li>
  </ul>
</form>
`;
